landingai/telemetry.py

```diff
@@ -0,0 +1,77 @@
+"""This module contains the telemetry configuration and APIs for the landingai package (intented for internal use only)."""
+
+
+import os
+import platform
+import sys
+from functools import lru_cache
+from importlib.metadata import version
+from pathlib import Path
+from typing import Dict
+
+
+@lru_cache(maxsize=None)
+def get_runtime_environment_info() -> Dict[str, str]:
+    """Return a set of runtime environment information in key value pairs."""
+    return {
+        "lib_type": "pylib",
+        "lib_version": version("landingai"),
+        "python_version": platform.python_version(),
+        "os": platform.platform(),
+        "runtime": _resolve_python_runtime(),
+    }
+
+
+@lru_cache(maxsize=None)
+def is_running_in_pytest() -> bool:
+    """Return True if the code is running in a pytest session."""
+    # See: https://stackoverflow.com/questions/25188119/test-if-code-is-executed-from-within-a-py-test-session
+    return "pytest" in sys.modules
+
+
+def _resolve_python_runtime() -> str:
+    if _is_running_in_colab():
+        runtime = "colab"
+    elif _is_running_in_notebook():
+        runtime = "notebook"
+    elif _is_running_in_streamlit():
+        runtime = "streamlit"
+    elif is_running_in_pytest():
+        runtime = "pytest"
+    else:
+        runtime = Path(os.environ.get("_", "unknown")).name
+    return runtime
+
+
+def _is_running_in_colab() -> bool:
+    """Return True if the code is running in a Google Colab notebook."""
+    try:
+        return get_ipython().__class__.__module__ == "google.colab._shell"  # type: ignore
+    except NameError:
+        return False  # Probably standard Python interpreter
+
+
+def _is_running_in_notebook() -> bool:
```

Comment:
I think these functions are more applicable than just telemetry...in fact today we are already doing 

- Jupyter detection https://vscode.dev/github/landing-ai/landingai-python/blob/matias/introduce-data-layer/landingai/common.py#L186
- Colab detection https://vscode.dev/github/landing-ai/landingai-python/blob/matias/introduce-data-layer/landingai/io.py#L128

Can you put these on common.py (and later we will merge them)?

Reply:
chatted offline, will refactor these code into a new `notebook_utils` module.

Reply:
https://github.com/landing-ai/landingai-python/pull/80